### PR TITLE
Arm64/Sve: Add SignExtend* and ZeroExtend* math APIs

### PR DESCRIPTION
**Pull request by kunalspathak**
- SignExtend16
- SignExtend32
- SignExtend8
- ZeroExtend16
- ZeroExtend32
- ZeroExtend8

All tests passing: https://gist.github.com/kunalspathak/d8c244f02f51e1a9e247f97f9d23b9e3

Contributes to https://github.com/dotnet/runtime/issues/99957

### CODE REVIEW

**Comment by kunalspathak** [comment]
@dotnet/arm64-contrib

**Review by TIHan — APPROVED** [review]
LGTM

File: src/coreclr/jit/hwintrinsiccodegenarm64.cpp [hunk 1]
```diff
@@ -448,14 +448,13 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                         if (intrin.op3->isContained())
                         {
                             assert(intrin.op3->IsVectorZero());
-                            if (intrin.op1->isContained())
+                            if (intrin.op1->isContained() || intrin.op1->IsMaskAllBitsSet())
```
Comment (tannergooding) [inline src/coreclr/jit/hwintrinsiccodegenarm64.cpp:451]: The `intrin.op1->isContained()` seems pointless here, no?

We just need to know its not contained for an instruction that actually requires its use (like `abs`)

Comment (kunalspathak) [reply src/coreclr/jit/hwintrinsiccodegenarm64.cpp:451]: i will have to double check on this one. There was lot of back and forth on this for certain scenarios. I will verify it on next round.

**Pull request merged**